### PR TITLE
:tada: New Destination: DynamoDB

### PR DESCRIPTION
**Pull request by jinnigu**
## What
Added AWS DynamoDB as the destination.
https://aws.amazon.com/dynamodb/

## Recommended reading order
1. `DynamodbDestination.java`
2. `DynamodbChecker.java`
3. `DynamodbConsumer.java`
4. `DynamodbWriter.java`
5. `DynamodbDestinationAcceptanceTest.java`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
thanks for the contribution @jinnig

File: airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbOutputTableHelper.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      paths.add(tableName);
+    }
+    if (namespace != null) {
+      paths.add(new ExtendedNameTransformer().convertStreamName(namespace));
+    }
+    if (streamName != null) {
+      paths.add(new ExtendedNameTransformer().convertStreamName(streamName));
+    }
+
+    return String.join("_", paths);
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbOutputTableHelper.java:51]: This should follow the convention from [here](https://docs.airbyte.io/understanding-airbyte/namespaces#examples).

The custom_namespace dont custom shouldn't start with the table name informed in spec.json. This is possible?
Screenshot from 2021-08-24 21-03-59

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
LGTM! Already test locally and use with the webapp and works great! Some small comments let me know what do you think about them @jinnig

File: airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbWriter.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        .withAttributeType(ScalarAttributeType.N);
+    KeySchemaElement partitionKeySchema = new KeySchemaElement()
+        .withAttributeName(JavaBaseConstants.COLUMN_NAME_AB_ID)
+        .withKeyType(KeyType.HASH);
+    KeySchemaElement sortKeySchema = new KeySchemaElement()
+        .withAttributeName("sync_time")
+        .withKeyType(KeyType.RANGE);
+    ProvisionedThroughput throughput = new ProvisionedThroughput()
+        .withReadCapacityUnits(10L)
+        .withWriteCapacityUnits(10L);
```
[the remark this answers is not in the document]
Comment (jinnig) [reply airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbWriter.java]: The change has been added in the new commit.

File: airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbOutputTableHelper.java (hunk 1 above)
Comment (jinnig) [reply airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbOutputTableHelper.java:51]: We changed the sequence for the table name in the new commit.

File: airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbWriter.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    ObjectMapper mapper = new ObjectMapper();
+    Map<String, Object> dataMap = mapper.convertValue(recordMessage.getData(), new TypeReference<Map<String, Object>>() {});
+
+    var item = new Item()
+        .withPrimaryKey(JavaBaseConstants.COLUMN_NAME_AB_ID, UUID.randomUUID().toString(), "sync_time", uploadTimestamp)
+        .withMap(JavaBaseConstants.COLUMN_NAME_DATA, dataMap)
+        .withLong(JavaBaseConstants.COLUMN_NAME_EMITTED_AT, recordMessage.getEmittedAt());
+    tableWriteItems.addItemToPut(item);
+    BatchWriteItemOutcome outcome;
+    if (tableWriteItems.getItemsToPut().size() >= 25) {
```
[the remark this answers is not in the document]
Comment (jinnig) [reply airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbWriter.java]: Sure. Please take a look in the following commit.

File: airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbChecker.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        Arrays.asList(new KeySchemaElement(JavaBaseConstants.COLUMN_NAME_AB_ID, KeyType.HASH), new KeySchemaElement("sync_time", KeyType.RANGE)),
+        Arrays.asList(new AttributeDefinition(JavaBaseConstants.COLUMN_NAME_AB_ID, ScalarAttributeType.S),
+            new AttributeDefinition("sync_time", ScalarAttributeType.N)),
+        new ProvisionedThroughput(1L, 1L));
+    table.waitForActive();
+
+    try {
+      PutItemOutcome outcome = table
+          .putItem(
+              new Item().withPrimaryKey(JavaBaseConstants.COLUMN_NAME_AB_ID, UUID.randomUUID().toString(), "sync_time", System.currentTimeMillis()));
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-dynamodb/src/main/java/io/airbyte/integrations/destination/dynamodb/DynamodbChecker.java:64]: @jinnig Dynamodb is a key-value and document database. Using the primary key as a hash type means you're focusing this connector as a document database? Or a hash type PK using in key-value for Dynamodb is not a problem?

**Review by marcosmarxm — APPROVED** [review]
LGMT @jinnig awesome work!

**Pull request merged**